ApiDemos/kotlin/app/src/main/java/com/example/kotlindemos/CameraDemoActivity.kt

```diff
@@ -0,0 +1,356 @@
+            target(LatLng(-33.87365, 151.20689))
+            .zoom(15.5f)
+            .bearing(0f)
+            .tilt(25f)
+            .build()
+
+
+    private lateinit var map: GoogleMap
+
+    private lateinit var animateToggle: CompoundButton
+    private lateinit var customDurationToggle: CompoundButton
+    private lateinit var customDurationBar: SeekBar
+    private var currPolylineOptions: PolylineOptions? = null
+    private var isCanceled = false
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.camera_demo)
+
+        animateToggle = findViewById(R.id.animate)
+        customDurationToggle = findViewById(R.id.duration_toggle)
+        customDurationBar = findViewById(R.id.duration_bar)
+
+        updateEnabledState()
+
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        updateEnabledState()
+    }
+
+    override fun onMapReady(googleMap: GoogleMap?) {
+
+        // return early if the map was not initialised properly
+        map = googleMap ?: return
+
+        with(googleMap) {
+            setOnCameraIdleListener(this@CameraDemoActivity)
+            setOnCameraMoveStartedListener(this@CameraDemoActivity)
+            setOnCameraMoveListener(this@CameraDemoActivity)
+            setOnCameraMoveCanceledListener(this@CameraDemoActivity)
+
+            // We will provide our own zoom controls.
+            uiSettings.isZoomControlsEnabled = false
+            uiSettings.isMyLocationButtonEnabled = true
+
+            // Show Sydney
+            moveCamera(CameraUpdateFactory.newLatLngZoom(sydneyLatLng, 10f))
+        }
+    }
+
+    /**
+     * When the map is not ready the CameraUpdateFactory cannot be used. This should be used to wrap
+     * all entry points that call methods on the Google Maps API.
+     *
+     * @param stuffToDo the code to be executed if the map is initialised
+     */
+    private fun checkReadyThen(stuffToDo: () -> Unit) {
+        if (!::map.isInitialized) {
+            Toast.makeText(this, R.string.map_not_ready, Toast.LENGTH_SHORT).show()
+        } else {
+            stuffToDo()
+        }
+    }
+
+    /**
+     * Called when the Go To Bondi button is clicked.
+     */
+    @Suppress("UNUSED_PARAMETER")
+    fun onGoToBondi(view: View) {
+        checkReadyThen {
+            changeCamera(CameraUpdateFactory.newCameraPosition(bondiLocation))
+        }
+    }
+
+    /**
+     * Called when the Animate To Sydney button is clicked.
+     */
+    @Suppress("UNUSED_PARAMETER")
+    fun onGoToSydney(view: View) {
+        checkReadyThen {
+            changeCamera(CameraUpdateFactory.newCameraPosition(sydneyLocation),
+                    object : CancelableCallback {
+                override fun onFinish() {
```

Comment:
I think the indentation here needs to be fixed

Reply:
Fixed!